File: src/static/js/kml.js


// render polygon to kml
google.maps.Polygon.prototype.kml = function() {
    var kml = "<Polygon>";
    kml+= "<outerBoundaryIs>";
    kml+= "<LinearRing>";
    kml+="<coordinates>";
    this.getPath().forEach(function(p) {
        kml += p.lng() + "," + p.lat() + ",0\n";
    });
    var p = this.getPath()[0];
    kml += p.lng() + "," + p.lat() + ",0\n";
    kml += "</coordinates>";
    kml+= "</LinearRing>";
    kml+= "</outerBoundaryIs>";
    kml += "</Polygon>";
    return kml;
};

var KML = {
   path_to_kml: function(path) {
        var kml = "<Polygon>";
        kml+= "<outerBoundaryIs>";
        kml+= "<LinearRing>";
        kml+="<coordinates>";
        _.each(path, function(p) {
            kml += p.lng() + "," + p.lat() + ",0\n";
        });
        // close polygon
        var p = path[0];
        kml += p.lng() + "," + p.lat() + ",0\n";
        kml += "</coordinates>";
        kml += "</LinearRing>";
        kml += "</outerBoundaryIs>";
        kml += "</Polygon>";
        return kml;
   }
};

File: src/application/kml.py


def path_to_kml(paths):
    kml = "<Polygon>"
    kml+= "<outerBoundaryIs>"
    kml+= "<LinearRing>"
    kml+="<coordinates>"
    for p in paths[0]:
        kml+= str(p[1]) + "," + str(p[0]) + ",0\n"
    kml += "</coordinates>"
    kml+= "</LinearRing>"
    kml+= "</outerBoundaryIs>"
    for inner in paths[1:]:
        kml +="<innerBoundaryIs>"
        kml+= "<LinearRing>"
        kml+="<coordinates>"
        for p in inner:
            kml+= str(p[1]) + "," + str(p[0]) + ",0\n"
        kml += "</coordinates>"
        kml+= "</LinearRing>"
        kml +="</innerBoundaryIs>"
    kml += "</Polygon>"
    return kml;
